Description: Disable examples and unavailable deps
 termwiz is proper unavailable,
 font8x8 is cruft,
 color-eyre (many), fakeit (1) and octocrab (1) are used by examples.
 Examples are too many and drastically slow down builds and autopkgtest runs.
Forwarded: not-needed
Last-Update: 2024-08-31
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,227 +67,227 @@
-[[example]]
-name = "async"
-path = "examples/async.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "barchart"
-path = "examples/barchart.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "barchart-grouped"
-path = "examples/barchart-grouped.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "block"
-path = "examples/block.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "calendar"
-path = "examples/calendar.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "widget-calendar",
-]
-
-[[example]]
-name = "canvas"
-path = "examples/canvas.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "chart"
-path = "examples/chart.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "colors"
-path = "examples/colors.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "colors_rgb"
-path = "examples/colors_rgb.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "palette",
-]
-
-[[example]]
-name = "constraint-explorer"
-path = "examples/constraint-explorer.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "constraints"
-path = "examples/constraints.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "custom_widget"
-path = "examples/custom_widget.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "demo"
-path = "examples/demo/main.rs"
-doc-scrape-examples = false
-
-[[example]]
-name = "demo2"
-path = "examples/demo2/main.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "palette",
-    "widget-calendar",
-]
-
-[[example]]
-name = "docsrs"
-path = "examples/docsrs.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "flex"
-path = "examples/flex.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "gauge"
-path = "examples/gauge.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "hello_world"
-path = "examples/hello_world.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "hyperlink"
-path = "examples/hyperlink.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "inline"
-path = "examples/inline.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "layout"
-path = "examples/layout.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "line_gauge"
-path = "examples/line_gauge.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "list"
-path = "examples/list.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "minimal"
-path = "examples/minimal.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "modifiers"
-path = "examples/modifiers.rs"
-doc-scrape-examples = false
-required-features = ["crossterm"]
-
-[[example]]
-name = "panic"
-path = "examples/panic.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "paragraph"
-path = "examples/paragraph.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "popup"
-path = "examples/popup.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "ratatui-logo"
-path = "examples/ratatui-logo.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "scrollbar"
-path = "examples/scrollbar.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "sparkline"
-path = "examples/sparkline.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "table"
-path = "examples/table.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "tabs"
-path = "examples/tabs.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "tracing"
-path = "examples/tracing.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "user_input"
-path = "examples/user_input.rs"
-doc-scrape-examples = true
-required-features = ["crossterm"]
-
-[[example]]
-name = "widget_impl"
-path = "examples/widget_impl.rs"
-doc-scrape-examples = true
-required-features = [
-    "crossterm",
-    "unstable-widget-ref",
-]
+#[[example]]
+#name = "async"
+#path = "examples/async.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "barchart"
+#path = "examples/barchart.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "barchart-grouped"
+#path = "examples/barchart-grouped.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "block"
+#path = "examples/block.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "calendar"
+#path = "examples/calendar.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "widget-calendar",
+#]
+
+#[[example]]
+#name = "canvas"
+#path = "examples/canvas.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "chart"
+#path = "examples/chart.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "colors"
+#path = "examples/colors.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "colors_rgb"
+#path = "examples/colors_rgb.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "palette",
+#]
+
+#[[example]]
+#name = "constraint-explorer"
+#path = "examples/constraint-explorer.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "constraints"
+#path = "examples/constraints.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "custom_widget"
+#path = "examples/custom_widget.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "demo"
+#path = "examples/demo/main.rs"
+#doc-scrape-examples = false
+
+#[[example]]
+#name = "demo2"
+#path = "examples/demo2/main.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "palette",
+#    "widget-calendar",
+#]
+
+#[[example]]
+#name = "docsrs"
+#path = "examples/docsrs.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "flex"
+#path = "examples/flex.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "gauge"
+#path = "examples/gauge.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "hello_world"
+#path = "examples/hello_world.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "hyperlink"
+#path = "examples/hyperlink.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "inline"
+#path = "examples/inline.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "layout"
+#path = "examples/layout.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "line_gauge"
+#path = "examples/line_gauge.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "list"
+#path = "examples/list.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "minimal"
+#path = "examples/minimal.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "modifiers"
+#path = "examples/modifiers.rs"
+#doc-scrape-examples = false
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "panic"
+#path = "examples/panic.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "paragraph"
+#path = "examples/paragraph.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "popup"
+#path = "examples/popup.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "ratatui-logo"
+#path = "examples/ratatui-logo.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "scrollbar"
+#path = "examples/scrollbar.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "sparkline"
+#path = "examples/sparkline.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "table"
+#path = "examples/table.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "tabs"
+#path = "examples/tabs.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "tracing"
+#path = "examples/tracing.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "user_input"
+#path = "examples/user_input.rs"
+#doc-scrape-examples = true
+#required-features = ["crossterm"]
+
+#[[example]]
+#name = "widget_impl"
+#path = "examples/widget_impl.rs"
+#doc-scrape-examples = true
+#required-features = [
+#    "crossterm",
+#    "unstable-widget-ref",
+#]
@@ -352,4 +352,4 @@
-[[bench]]
-name = "main"
-path = "benches/main.rs"
-harness = false
+#[[bench]]
+#name = "main"
+#path = "benches/main.rs"
+#harness = false
@@ -402,3 +402,3 @@
-[dependencies.termwiz]
-version = "0.22.0"
-optional = true
+#[dependencies.termwiz]
+#version = "0.22.0"
+#optional = true
@@ -423,2 +423,2 @@
-[dev-dependencies.color-eyre]
-version = "0.6.2"
+#[dev-dependencies.color-eyre]
+#version = "0.6.2"
@@ -437,2 +437,2 @@
-[dev-dependencies.fakeit]
-version = "1.1"
+#[dev-dependencies.fakeit]
+#version = "1.1"
@@ -440,2 +440,2 @@
-[dev-dependencies.font8x8]
-version = "0.3.1"
+#[dev-dependencies.font8x8]
+#version = "0.3.1"
@@ -449,2 +449,2 @@
-[dev-dependencies.octocrab]
-version = "0.39.0"
+#[dev-dependencies.octocrab]
+#version = "0.39.0"
@@ -501 +501 @@
-termwiz = ["dep:termwiz"]
+#termwiz = ["dep:termwiz"]
